src/QSwagGenerator/Generators/SchemaGenerator.cs

```diff
@@ -109,11 +109,17 @@ internal SchemaObject MapToSchema(JSchema jSchema, HashSet<Uri> processedDefinit
                 foreach (var keyValuePair in jSchema.Properties)
                 {
                     var property = keyValuePair.Value;
-                    if (processedDefinitions.Contains(property.Id))
-                        schema.Properties.Add(keyValuePair.Key, new SchemaObject {Ref = $"#/definitions/{property.Id}"});
-                    else if (property.Type != null && property.Type.Value.HasFlag(JSchemaType.Array) && processedDefinitions.Contains(property.Items.First().Id))
-                        schema.Properties.Add(keyValuePair.Key, new SchemaObject{Type=SchemaType.Array,
-                                Items = new List<SchemaObject> {new SchemaObject { Ref = $"#/definitions/{property.Items.First().Id}" } } });
+                    if (property.Type != null && property.Type.Value.HasFlag(JSchemaType.Array) &&
+                        processedDefinitions.Contains(property.Items.First().Id))
```

Comment:
What if collection is empty? Re-write in reverse. `property.Items.Any(item=>processDefiniton.Contains(item.Id))`